apps/block_scout_web/lib/block_scout_web/controllers/address_validation_controller.ex

```diff
@@ -67,6 +67,8 @@ defmodule BlockScoutWeb.AddressValidationController do
   def index(conn, %{"address_id" => address_hash_string}) do
     with {:ok, address_hash} <- Chain.string_to_address_hash(address_hash_string),
          {:ok, address} <- Chain.find_or_insert_address_from_hash(address_hash) do
+      AddressContractVerificationController.check_sourcify(address_hash_string, conn)
```

Comment:
... and here. Otherwise, for the worst case scenario when a contract is unverified (that would be the most often scenario as well) we'll end up making these requests every time.